cakephpsphinx/themes/cakephp/static/js/header.js

```diff
@@ -39,20 +40,14 @@ App.Header = (function () {
             didScroll = true;
         });
 
-        // Debounce the header toggling to ever 250ms
+        // Debounce the header toggling to every 250ms
         var toggleHeader = function () {
             if (didScroll) {
                 hasScrolled();
                 didScroll = false;
             }
-            setTimeout(toggleHeader, 250);
         };
-        setTimeout(toggleHeader, 250);
-
-        // If we're directly linking to a section, hide the nav.
-        if (window.location.hash.length) {
-            $header.addClass('nav-up');
-        }
+        setInterval(toggleHeader, 250);
```

Comment:
If the page is 'busy' interval instances can pile up. Using setTimeout() in a callback ensure you get at most one timer trigger every 250ms. It might not matter here though.